contracts/profit/src/state/mod.rs

```diff
@@ -36,13 +40,33 @@ const STATE: Item<'static, State> = Item::new("contract_state");
 
 type IcaConnector = dex::IcaConnector<OpenIca, ContractResult<DexResponse<Idle>>>;
 
+fn on_config_update(new_config: &Config, now: Timestamp) -> ContractResult<PlatformResponse> {
+    new_config
+        .time_alarms()
+        .setup_alarm(now + Duration::from_hours(new_config.cadence_hours()))
+        .map(PlatformResponse::messages_only)
+        .map_err(Into::into)
+}
+
 pub(crate) trait ConfigManagement
 where
     Self: Sized,
 {
-    fn try_update_config(self, cadence_hours: CadenceHours) -> ContractResult<Self>;
+    fn try_update_config(
+        self,
+        _: Timestamp,
+        _: CadenceHours,
+    ) -> ContractResult<StateMachineResponse<Self>> {
+        Err(ContractError::UnsupportedOperation(String::from(
+            "Configuration changes are not allowed in this state.",
+        )))
+    }
 
-    fn try_query_config(&self) -> ContractResult<ConfigResponse>;
+    fn try_query_config(&self) -> ContractResult<ConfigResponse> {
+        Err(ContractError::UnsupportedOperation(String::from(
+            "Configuration querying is not allowed in this state.",
+        )))
+    }
```

Comment:
looking into the code, I realized that it's not correct to refuse config queries. Therefore please remove the default implementation and provide one at each of the four states.

Reply:
I can't really do that for all four. I can do it only for the two where I have overridden it.